addons/mail/static/src/models/message_sender/message_sender.js

```diff
@@ -0,0 +1,272 @@
+                    command: command.name,
+                    postData,
+                }));
+            } else {
+                messageId = await this.async(() =>
+                    this.env.models['mail.thread'].performRpcMessagePost({
+                        postData,
+                        threadId: this.thread.id,
+                        threadModel: this.thread.model,
+                    })
+                );
+            }
+            return messageId;
+        }
+
+        async _postMailThread(postData) {
+            Object.assign(postData, {
+                subtype_xmlid: this.thread.composer.isLog ? 'mail.mt_note' : 'mail.mt_comment',
+            });
+            if (!this.thread.composer.isLog) {
+                postData.context = {
+                    mail_post_autofollow: true,
+                };
+            }
+            const messageId = await this.async(() =>
+                this.env.models['mail.thread'].performRpcMessagePost({
+                    postData,
+                    threadId: this.thread.id,
+                    threadModel: this.thread.model,
+                })
+            );
+            const [messageData] = await this.async(() => this.env.services.rpc({
+                model: 'mail.message',
+                method: 'message_format',
+                args: [[messageId]],
+            }, { shadow: true }));
+            this.env.models['mail.message'].insert(Object.assign(
+                {},
+                this.env.models['mail.message'].convertData(messageData),
+                {
+                    originThread: [['insert', {
+                        id: this.thread.id,
+                        model: this.thread.model,
+                    }]],
+                })
+            );
+            this.thread.loadNewMessages();
+        }
+
+        /**
+         * Function that process the message stored to be sent.
+         */
+        async _processMessagesToBeSent() {
+            if (this.isSendingMessages) {
+                return;
+            }
+            this.update({
+                isSendingMessages: true,
+            });
+            const messages = this.pendingMessagesToBeSent.filter((message) => !message.hasSendError);
+            const message = messages[0];
+            try {
+                if (message) {
+                    let postData = {
+                        attachment_ids: message.attachments.map(attachement => attachement.id),
+                        body: message.body,
+                        message_type: 'comment',
+                        partner_ids: message.partner_ids,
+                        subject: message.subject,
+                    };
+                    if (this.thread.model === 'mail.channel') {
+                        await this._postMailChannel(postData);
+                    } else {
+                        await this._postMailThread(postData);
+                    }
+                    message.delete();
```

Comment:
add a comment to explain, because conceptually the message should not be deleted after it is posted, but it should just be updated with server information